### PR TITLE
reuse legacy option to decide the behavior of link resolve

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: src/docfx/lib/HtmlUtility.cs [hunk 1]
```diff
@@ -170,7 +170,7 @@ private static void AddLinkType(this HtmlNode html, string tag, string attribute
                 }
                 if (href[0] == '/' || href[0] == '\\')
                 {
-                    node.SetAttributeValue(attribute, AddLocaleIfMissing(HrefToLower(href), locale));
+                    node.SetAttributeValue(attribute, AddLocaleIfMissing(shouldToLowerCase ? HrefToLower(href) : href, locale));
```
Comment (yufeih) [inline src/docfx/lib/HtmlUtility.cs]: the `ToLower` was for the purpose of v2 backward compact, can you examine why some of the URL's are not converted to lower case in v2?

File: src/docfx/config/Config.cs [hunk 2]
```diff
@@ -48,6 +48,11 @@ internal class Config
         /// </summary>
         public readonly JObject GlobalMetadata = new JObject();
 
+        /// <summary>
+        /// Just for backward compatibility, Indicate that whether enable legacy output.
+        /// </summary>
+        public readonly bool Legacy = false;
```
[the remark this answers is not in the document]
Comment (928PJY) [reply src/docfx/config/Config.cs]: it already applied.
[https://github.com/dotnet/docfx/blob/v3/src/docfx/build/Build.cs#L36](https://github.com/dotnet/docfx/blob/v3/src/docfx/build/Build.cs#L36)

---
In reply to: [212530895](https://github.com/dotnet/docfx/pull/3274#discussion_r212530895) [](ancestors = 212530895)

File: src/docfx/build/resolve/Resolve.cs [hunk 3]
```diff
@@ -43,6 +43,15 @@ public static (Error error, string href, string fragment, Document file) TryReso
             // Self reference, don't build the file, leave href as is
             if (file == relativeTo)
             {
+                if (relativeTo.Docset.Config.Legacy)
```
[the remark this answers is not in the document]
Comment (928PJY) [reply src/docfx/build/resolve/Resolve.cs]: why?

---
In reply to: [212530732](https://github.com/dotnet/docfx/pull/3274#discussion_r212530732) [](ancestors = 212530732)

File: src/docfx/config/Config.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (928PJY) [reply src/docfx/config/Config.cs]: move legacy to docset.

---
In reply to: [212852541](https://github.com/dotnet/docfx/pull/3274#discussion_r212852541) [](ancestors = 212852541)

File: src/docfx/lib/HtmlUtility.cs (hunk 1 above)
Comment (928PJY) [reply src/docfx/lib/HtmlUtility.cs]: we will leave this issue, add a TODO

---
In reply to: [212531773](https://github.com/dotnet/docfx/pull/3274#discussion_r212531773) [](ancestors = 212531773)

**Pull request merged**